### PR TITLE
fix: convert to ReadableSize & Durations

### PR DESCRIPTION
**Pull request by masonyc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Convert to u64 to ReadableSize and Durations

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
[#2582](https://github.com/GreptimeTeam/greptimedb/issues/2582)

### CODE REVIEW

File: src/datanode/src/heartbeat.rs [hunk 1]
```diff
@@ -332,14 +332,14 @@ pub async fn new_metasrv_client(
     let member_id = node_id;
 
     let config = ChannelConfig::new()
-        .timeout(Duration::from_millis(meta_config.timeout_millis))
-        .connect_timeout(Duration::from_millis(meta_config.connect_timeout_millis))
+        .timeout(meta_config.timeout_millis)
```
[the remark this answers is not in the document]
Comment (masonyc) [reply src/datanode/src/heartbeat.rs]: thanks @killme2008 , removed postfix on heartbeat options as well

File: config/datanode.example.toml [hunk 2]
```diff
@@ -13,19 +13,19 @@ rpc_runtime_size = 8
 require_lease_before_startup = false
 
 [heartbeat]
-# Interval for sending heartbeat messages to the Metasrv in milliseconds, 3000 by default.
-interval_millis = 3000
+# Interval for sending heartbeat messages to the Metasrv, 3000 milliseconds by default.
+interval = "3000ms"
```
Comment (killme2008) [inline config/datanode.example.toml:17]:
```suggestion
interval = "3s"
```

File: src/cmd/src/options.rs [hunk 3]
```diff
@@ -110,6 +110,12 @@ impl Options {
         if let Some(config_file) = config_file {
             layered_config = layered_config.add_source(File::new(config_file, FileFormat::Toml));
         }
+        let oapts = layered_config
+            .clone()
+            .build()
+            .context(LoadLayeredConfigSnafu)?;
+
+        dbg!(oapts);
```
Comment (killme2008) [inline src/cmd/src/options.rs:118]: Looks like it's unneeded. Just for debug purposes.

File: src/datanode/src/heartbeat.rs [hunk 4]
```diff
@@ -69,7 +69,7 @@ impl HeartbeatTask {
     ) -> Result<Self> {
         let region_alive_keeper = Arc::new(RegionAliveKeeper::new(
             region_server.clone(),
-            opts.heartbeat.interval_millis,
+            opts.heartbeat.interval.as_secs(),
```
Comment (killme2008) [inline src/datanode/src/heartbeat.rs:72]:
```suggestion
            opts.heartbeat.interval.as_millis(),
```

File: src/datanode/src/heartbeat.rs [hunk 5]
```diff
@@ -86,7 +86,7 @@ impl HeartbeatTask {
             running: Arc::new(AtomicBool::new(false)),
             meta_client: Arc::new(meta_client),
             region_server,
-            interval: opts.heartbeat.interval_millis,
+            interval: opts.heartbeat.interval.as_secs(),
```
Comment (killme2008) [inline src/datanode/src/heartbeat.rs:89]:
```suggestion
            interval: opts.heartbeat.interval.as_millis(),
```

File: src/frontend/src/heartbeat.rs [hunk 6]
```diff
@@ -49,8 +49,8 @@ impl HeartbeatTask {
     ) -> Self {
         HeartbeatTask {
             meta_client,
-            report_interval: heartbeat_opts.interval_millis,
-            retry_interval: heartbeat_opts.retry_interval_millis,
+            report_interval: heartbeat_opts.interval.as_secs(),
```
Comment (killme2008) [inline src/frontend/src/heartbeat.rs:52]:
```suggestion
            report_interval: heartbeat_opts.interval.as_millis(),
```

File: src/frontend/src/heartbeat.rs [hunk 7]
```diff
@@ -49,8 +49,8 @@ impl HeartbeatTask {
...
+            retry_interval: heartbeat_opts.retry_interval.as_secs(),
```
Comment (killme2008) [inline src/frontend/src/heartbeat.rs:53]:
```suggestion
            retry_interval: heartbeat_opts.retry_interval.as_millis(),
```

File: src/meta-client/src/lib.rs [hunk 8]
```diff
@@ -21,31 +23,35 @@ pub mod error;
...
         Self {
             metasrv_addrs: vec!["127.0.0.1:3002".to_string()],
-            timeout_millis: 3_000u64,
-            heartbeat_timeout_millis: default_heartbeat_timeout_millis(),
-            ddl_timeout_millis: default_ddl_timeout_millis(),
-            connect_timeout_millis: 1_000u64,
+            timeout: Duration::from_millis(3_000u64),
+            heartbeat_timeout: default_heartbeat_timeout(),
+            ddl_timeout: default_ddl_timeout(),
+            connect_timeout: Duration::from_millis(1_000u64),
```
Comment (zhongzc) [inline src/meta-client/src/lib.rs:54]: It is recommended to uniformly adopt the form of default function

Comment (masonyc) [reply src/meta-client/src/lib.rs]: thanks @zhongzc , updated in the latest commit

File: src/frontend/src/heartbeat.rs (hunk 7 above)
Comment (masonyc) [reply src/frontend/src/heartbeat.rs]: thanks @killme2008 , updated in the latest commit

File: src/meta-client/src/lib.rs [hunk 9]
```diff
@@ -21,31 +23,43 @@ pub mod error;
...
+    #[serde(with = "humantime_serde")]
+    pub timeout: Duration,
+    #[serde(default = "default_heartbeat_timeout")]
+    #[serde(with = "humantime_serde")]
+    pub heartbeat_timeout: Duration,
+    #[serde(default = "default_ddl_timeout")]
+    #[serde(with = "humantime_serde")]
+    pub ddl_timeout: Duration,
+    #[serde(with = "humantime_serde")]
+    pub connect_timeout: Duration,
```
Comment (zhongzc) [inline src/meta-client/src/lib.rs:35]:
```suggestion
    #[serde(default = "default_timeout")]
    #[serde(with = "humantime_serde")]
    pub timeout: Duration,
    #[serde(default = "default_heartbeat_timeout")]
    #[serde(with = "humantime_serde")]
    pub heartbeat_timeout: Duration,
    #[serde(default = "default_ddl_timeout")]
    #[serde(with = "humantime_serde")]
    pub ddl_timeout: Duration,
    #[serde(default = "default_connect_timeout")]
    #[serde(with = "humantime_serde")]
    pub connect_timeout: Duration,
```

**Review by zhongzc — APPROVED** [review]
Rest LGTM

**Review by killme2008 — APPROVED** [review]

**Pull request merged**